connectors/sources/jira.py

```diff
@@ -0,0 +1,467 @@
+            timeout=timeout,
+            raise_for_status=True,
+        )
+
+    async def close(self):
+        """Closes unclosed client session"""
+        if self.session is None:
+            return
+        await self.session.close()
+
+    async def _api_call(self, url_name, is_pagination=False, **url_kwargs):
+        """Method for adding retries whenever exception raised during an api calls
+
+        Args:
+            url_name (str): Jira query name to be executed.
+            is_pagination (bool): Flag to check if the pagination is required
+            url_kwargs (dict): Url kwargs to format the query.
+
+        Raises:
+            exception: An instance of an exception class.
+
+        Yields:
+            response: Return api response.
+        """
+        retry = 0
+        start_at = 0
+        url = parse.urljoin(self.host_url, URLS[url_name].format(**url_kwargs))
+
+        while True:
+            try:
+                async with self.session.get(
+                    url=url,
+                    ssl=self.ssl_ctx,
+                ) as response:
+                    if is_pagination:
+                        response_json = await response.json()
+                        total = response_json.get("total")
+                        yield response_json
+                        if start_at + FETCH_SIZE >= total or total <= FETCH_SIZE:
+                            break
+                        start_at += FETCH_SIZE
+                        url = parse.urljoin(
+                            self.host_url,
+                            URLS[url_name].format(
+                                startAt=start_at, maxResults=FETCH_SIZE
+                            ),
+                        )
+                        retry = 0
+                    else:
+                        yield response
+                        break
+            except Exception as exception:
+                if isinstance(
+                    exception, aiohttp.client_exceptions.ServerDisconnectedError
+                ):
+                    await self.close()
+                    self.session = self._generate_session()
+                retry += 1
+                if retry > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                await asyncio.sleep(RETRY_INTERVAL**retry)
+
+    async def get_content(self, issue_key, attachment, timestamp=None, doit=False):
+        """Extracts the content for allowed file types.
+
+        Args:
+            issue_key (str): Issue key to generate `_id` for attachment document
+            attachment (dictionary): Formatted attachment document.
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
```

Comment:
Why have this flag?

Reply:
This flag is used to manage the already indexed attachments. If the attachment has been updated then the content will get extracted (`doit` is True at that time) since the _timestamp has also updated for that attachment else `doit` remains False and content will not get extract.